### PR TITLE
Jv remove kernel module collection

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

See https://github.com/stackrox/collector/issues/1107

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Confirmed that it is not possible to use kernel module as a collection method

## Testing Performed

See above

### CODE REVIEW

**Review by 0x656b694d — APPROVED** [review]
LGTM

File: operator/apis/platform/v1alpha1/securedcluster_types.go [hunk 1]
```diff
@@ -190,17 +190,15 @@ type PerNodeSpec struct {
...
+// +kubebuilder:validation:Enum=EBPF;CORE_BPF;NoCollection
 type CollectionMethod string
 
 const (
 	// CollectionEBPF means: use EBPF collection.
 	CollectionEBPF CollectionMethod = "EBPF"
 	// CollectionCOREBPF means: use CORE_BPF collection.
 	CollectionCOREBPF CollectionMethod = "CORE_BPF"
-	// CollectionKernelModule means: use KERNEL_MODULE collection.
-	CollectionKernelModule CollectionMethod = "KernelModule"
```
Comment (SimonBaeumer) [inline operator/apis/platform/v1alpha1/securedcluster_types.go:203]: What happens if a customer still has "KernelModule" configured?
I am concerned that we introduce a backwards incompatible change which breaks customer instances.

**Review by SimonBaeumer — CHANGES_REQUESTED** [review]

File: operator/pkg/securedcluster/values/translation/translation.go [hunk 2]
```diff
@@ -275,8 +275,6 @@ func (t Translator) getCollectorContainerValues(collectorContainerSpec *platform
 		switch *c {
 		case platform.CollectionEBPF:
 			cv.SetStringValue("collectionMethod", storage.CollectionMethod_EBPF.String())
-		case platform.CollectionKernelModule:
```
Comment (kurlov) [inline operator/pkg/securedcluster/values/translation/translation.go:278]: I have a concern about backward capability in the operator.
Maybe we can keep `CollectionKernelModule` and mark it as deprecated in the comment. Also map `CollectionKernelModule` to `EBPF`. Then delete it in 4.x version.
Not sure about such approach though

**Review by dhaus67 — COMMENTED** [review]
Irrespective of the concerns mentioned for the operator, it'd be nice to add the changelog entry to mark the kernel module collection method now as removed.

File: operator/apis/platform/v1alpha1/securedcluster_types.go (hunk 1 above)
Comment (JoukoVirtanen) [reply operator/apis/platform/v1alpha1/securedcluster_types.go:203]: See my response to Aleksandr Kurlov.

File: roxctl/sensor/generate/collection_method.go [hunk 3]
```diff
@@ -9,11 +9,10 @@ import (
 
 var (
 	humanReadableToEnum = map[string]storage.CollectionMethod{
-		"default":       storage.CollectionMethod_UNSET_COLLECTION,
-		"none":          storage.CollectionMethod_NO_COLLECTION,
-		"kernel-module": storage.CollectionMethod_KERNEL_MODULE,
```
Comment (JoukoVirtanen) [inline roxctl/sensor/generate/collection_method.go:14]: Change to "ebpf"?

File: proto/storage/proto.lock [hunk 4]
```diff
@@ -1326,10 +1326,6 @@
                 "name": "NO_COLLECTION",
                 "integer": 1
               },
-              {
-                "name": "KERNEL_MODULE",
-                "integer": 2
-              },
```
Comment (JoukoVirtanen) [inline proto/storage/proto.lock:1332]: Keep?

File: proto/storage/cluster.proto [hunk 5]
```diff
@@ -55,7 +55,6 @@ message OrchestratorMetadata {
 enum CollectionMethod {
     UNSET_COLLECTION = 0;
     NO_COLLECTION    = 1;
-    KERNEL_MODULE    = 2;
```
Comment (JoukoVirtanen) [inline proto/storage/cluster.proto:58]: Keep?

File: ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx [hunk 6]
```diff
@@ -23,9 +23,9 @@ export const runtimeOptions = [
         value: 'NO_COLLECTION',
     },
     {
-        label: 'Kernel Module',
-        tableDisplay: 'Kernel Module',
-        value: 'KERNEL_MODULE',
+        label: 'CORE BPF',
+        tableDisplay: 'CORE BPF',
+        value: 'CORE_BPF',
```
Comment (JoukoVirtanen) [inline ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx:28]: Forgot to add core bpf here in an earlier PR.

File: ui/apps/platform/src/types/cluster.proto.ts [hunk 7]
```diff
@@ -49,7 +49,7 @@ export type ClusterOrchestratorMetadata = {
     apiVersions: string[];
 };
 
-export type CollectionMethod = 'UNSET_COLLECTION' | 'NO_COLLECTION' | 'KERNEL_MODULE' | 'EBPF';
+export type CollectionMethod = 'UNSET_COLLECTION' | 'NO_COLLECTION' | 'EBPF' | 'CORE_BPF';
```
Comment (JoukoVirtanen) [inline ui/apps/platform/src/types/cluster.proto.ts:52]: UI will need to be checked.

File: operator/pkg/securedcluster/values/translation/translation.go (hunk 2 above)
Comment (Stringy) [reply operator/pkg/securedcluster/values/translation/translation.go:278]: @kurlov - I've implemented a version of your idea where the collection method is still removed but we still check for the legacy value and switch to eBPF in this case. I've tested it locally and it seems to be working as expected, and I'm interested to hear what you think of my implementation

**Comment by Stringy** [comment]
I have tested operator and helm upgrades from ACS configured with kernel modules and everything seems to work. With the translation in place in the operator to turn kernel-module to eBPF collection, everything is very smooth.

With helm there's some caveats. It will switch the collectors to eBPF collection _unless_ the upgrade uses `--reuse-values`, which will instead force kernel-module collection despite it no longer being a valid value (looks like helm doesn't validate at all.) This is handled by the collector container itself, which switches to eBPF, so I think it's _probably_ fine, but I'm interested to hear what people think about it.

I believe roxctl upgrades will suffer from the same problem, but collector will also handle this and switch to eBPF. I'll try out roxctl upgrades today to confirm.

**Review by SimonBaeumer — APPROVED** [review]

File: ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx (hunk 6 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx:28]: Aha, looking with fresh eyes, `tableDisplay` is obsolete in `runtimeOptions` and `clusterTypeOptions`

UI team can follow up to delete after 4.1 release when no risk of merge conflicts.

**Comment by robbycochran** [comment]
We also need to remove the mention of kernel module in the roxctl description for collection method in `roxctl/sensor/generate/generate.go`

**Pull request merged**